clients/lsp-rust.el

```diff
@@ -928,5 +937,64 @@ them with `crate` or the crate name they refer to."
   (interactive)
   (lsp-find-locations "experimental/parentModule" nil :display-action display-action))
 
+(defun lsp-rust-analyzer-open-cargo-toml (&optional new-window)
+  "Open the closest Cargo.toml from the current file.
+
+   Rust-Analyzer LSP protocol documented here and added in November 2020
+   https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/lsp-extensions.md#open-cargotoml
+
+   If NEW-WINDOW (interactively the prefix argument) is non-nil,
+   open in a new window."
+  (interactive "P")
+  (-if-let (workspace (lsp-find-workspace 'rust-analyzer))
+      (-if-let* ((params (lsp-make-rust-analyzer-syntax-tree-params
+                          :text-document (lsp--text-document-identifier)))
+                 ;; REVIEW is there a convention how to define and call rust-analyzer
+                 ;; lsp extension methods?
+                 (cargo-toml-location (with-lsp-workspace workspace
+                                        (lsp-send-request (lsp-make-request
+                                                           "experimental/openCargoToml"
+                                                           params))))
+                 ;; REVIEW is there a more idiomatic to retrieve the :uri key
+                 ;; or default to nil?
+                 (cargo-toml-uri (gethash "uri" cargo-toml-location)))
```

Comment:
You should define new interface similar to what in `lsp-protocol` for server response, and use the generated getter instead of `gethash` for this

Reply:
am good to change it now - will need to work out how to define and use the protocol generator

Reply:
hey, i have been trying to define and use the protocol from looking at already implemented endpoints. However, I have also noticed that `lsp-rust-find-parent-module` doesn't use a protocol definition and instead relies on `lsp-find-locations`. 

Can you please clarify the most idiomatic way to handle this kind of `Response: Location | null`?

Reply:
Also, shouldn't it the method be called `lsp-rust-analyzer-find-parent-module` to be consistent with other functions?